### PR TITLE
feature/enhanced state survivor round6

### PR DESCRIPTION
**Pull request by ootakazuhiko**
- **test: guard enhanced state import edge cases**
- **docs,test: extend enhanced state import coverage**
- **test: enforce imported version upper bounds**
- **test: tighten enhanced-state-manager import fixtures**
- **test: harden enhanced-state import paths**

### CODE REVIEW

File: tests/unit/utils/enhanced-state-manager.test.ts [hunk 1]
```diff
@@ -582,39 +791,44 @@ describe('EnhancedStateManager helper behaviour', () => {
     const invalidArray: Array<number | string> = [80, 'not-a-number', 82];
 
     const numericKey = `legacy-buffer_${timestamp}`;
-    const invalidKey = `legacy-invalid_${timestamp}`;
-
-    await manager.importState({
-      metadata: { version: '1.0.0' },
+    const invalidKey = `legacy-invalid_${timestamp}`;    const exported = buildExportedState(manager, {
```
[the remark this answers is not in the document]
Comment (ootakazuhiko) [reply tests/unit/utils/enhanced-state-manager.test.ts]: Fixed by splitting the declaration across lines. Thanks for flagging it.
